fractal-theme/assets/js/src/search-results.js

```diff
@@ -0,0 +1,75 @@
+import Fuse from 'fuse.js'
+
+function fractalSearchResults() {
+  const searchComponent = document.querySelector('.search');
+  const searchInput = document.getElementById('searchPageInput');
+  const searchResults = document.getElementById('searchResults');
+  const SearchApi = '/assets/theme/js/search.json';
+
+  // Search options
+  const searchOptions = {
+    keys: [{
+      name: 'data.title',
+      weight: 0.5
+    }, {
+      name: 'data.primaryKeywords',
+      weight: 0.4
+    }, {
+      name: 'data.secondaryKeywords',
+      weight: 0.1
+    }]
+  };
+
+  // Fetch endpoint
+  fetch(SearchApi)
+    .then(response => {
+      return response.json()
+    })
+    // Search init
+    .then(search => {
+      const fuse = new Fuse(search, searchOptions)
+      const getSearchParams = new URLSearchParams(location.search);
+      const searchString = getSearchParams.get('q');
+      searchInput.value = searchString;
+      searchInput.focus();
+
+      // On load
+      const searchValueFuseTest = fuse.search(searchString);
+      searchValueFuseTest.forEach(element => {
+        searchResults.innerHTML += `
```

Comment:
Consider using insertAdjacentHTML instead.
The same applies to all cases below.
https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML